cmd/elasticsearch-operator/main.go

```diff
@@ -2,24 +2,111 @@ package main
 
 import (
 	"context"
+	"flag"
+	"fmt"
+	"os"
 	"runtime"
+	"strings"
 	"time"
 
 	stub "github.com/openshift/elasticsearch-operator/pkg/stub"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
 	sdk "github.com/operator-framework/operator-sdk/pkg/sdk"
 	k8sutil "github.com/operator-framework/operator-sdk/pkg/util/k8sutil"
 	sdkVersion "github.com/operator-framework/operator-sdk/version"
 
 	"github.com/sirupsen/logrus"
 )
 
+const (
+	// supported log formats
+	logFormatLogfmt = "logfmt"
+	logFormatJSON   = "json"
+
+	// supported log levels
```

Comment:
you should derive these from https://github.com/sirupsen/logrus/blob/master/logrus.go#L73 - and use the `Level.String` and `ParseLevel` methods to convert between string and integer.

Reply:
Do we want to use integers for log verbosity/severity?
That's vastly poorer user experience in comparison to human readable and fairly standard strings. Such as 'info' or 'debug'. Instead of '1' or '2'...

Reply:
we should use strings (for bonus points, accept both strings and integers) - but the strings are already defined in the logrus module, so we should not have to define them again here

Reply:
Those log levels are of type `uint32` in logrus: `type Level uint32
`.
I'm using `ParseLevel(lvl string) (Level, error)` to get internal representation (`Level`) from human readable string

What's the added value of using integers for configuring log-level in parallel with strings? I'm confused.

Reply:
Copy-paste from chat:

```go
func test(lvl string) (logrus.Level, error) {
    for _, logrusLevel := range logrus.AllLevels {
        if logrusLevel.String() == lvl {
            return logrusLevel, nil
        }
    }
    var l logrus.Level
    return l, fmt.Errorf("not a valid logrus Level: %s", lvl)
}
```

I didn't go this path because:
• O(N) for no reason
• I still need an array of strings, so I can print out help message what are supported level values